packages/state-manager/src/sagas/files/checkForMissingFiles/checkForMissingFiles.saga.ts

```diff
@@ -45,11 +44,13 @@ export function* checkForMissingFilesSaga(
               metadata: file
             }
           ])
-          return
+          continue
         }
 
         // Do not autodownload oversized files unless started manually
-        if (fileDownloadStatus?.downloadState !== DownloadState.Downloading && file.size > AUTODOWNLOAD_SIZE_LIMIT) return
+        if (fileDownloadStatus?.downloadState !== DownloadState.Downloading && file.size > AUTODOWNLOAD_SIZE_LIMIT) continue
+
+        if (fileDownloadStatus?.downloadState === DownloadState.Malicious || !file.size) continue
```

Comment:
I'd appreciate some comment above, like in the other 'if' statements in this saga, something like "Do not autodownload if the file was reported malicious or is missing reported file size".
By the way so far we were not preventing the file from download in this place but instead we were comparing reported size with the actual value in backend (which of course guarantees security). I'm wondering if adding this additional step will have any further consequences on the flow